libs/core/libs/shared/src/core_ops.rs

```diff
@@ -28,17 +28,21 @@ where
     T: TreeLike<F = SignedFile>,
 {
     // todo: revisit logic for what files can be finalized and how e.g. link substitutions, deleted files, files in pending shares, linked files
-    pub fn finalize(
+    fn decrypt(
         &mut self, id: &Uuid, account: &Account, public_key_cache: &mut LookupTable<Owner, String>,
     ) -> SharedResult<File> {
         let meta = self.find(id)?.clone();
         let file_type = meta.file_type();
-        let parent = *meta.parent();
         let last_modified = meta.timestamped_value.timestamp as u64;
         let name = self.name_using_links(id, account)?;
-        let id = *id;
+        let parent = *meta.parent();
         let last_modified_by = account.username.clone();
 
+        let id = match file_type {
```

Comment:
if we are directly getting the info of a link by id, it's id shouldn't change, right?

Reply:
I agree